test/prism_regression/defined.rb

```diff
@@ -0,0 +1,2 @@
+# typed: false
+defined?(Foo)
```

Comment:
In terms of our implementation these might not change much, because we just translate the value node, but as long as they're all valid uses then it can't hurt to have tests for them :)

Reply:
I didn't find tests that target `defined?` with different node types in `ruby/spec`, so I just added the above cases and I think we indeed don't need more special handling for them.

Reply:
Sorry I was blind and thought it's the same as your suggestion 😓